domain/lib/src/usecases/download_file/download_file_ios_interactor.dart

```diff
@@ -0,0 +1,69 @@
+// provided you comply with the Additional Terms applicable for LinShare software by
+// Linagora pursuant to Section 7 of the GNU Affero General Public License,
+// subsections (b), (c), and (e), pursuant to which you must notably (i) retain the
+// display in the interface of the “LinShare™” trademark/logo, the "Libre & Free" mention,
+// the words “You are using the Free and Open Source version of LinShare™, powered by
+// Linagora © 2009–2020. Contribute to Linshare R&D by subscribing to an Enterprise
+// offer!”. You must also retain the latter notice in all asynchronous messages such as
+// e-mails sent with the Program, (ii) retain all hypertext links between LinShare and
+// http://www.linshare.org, between linagora.com and Linagora, and (iii) refrain from
+// infringing Linagora intellectual property rights over its trademarks and commercial
+// brands. Other Additional Terms apply, see
+// <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf>
+// for more details.
+// This program is distributed in the hope that it will be useful, but WITHOUT ANY
+// WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+// FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for
+// more details.
+// You should have received a copy of the GNU Affero General Public License and its
+// applicable Additional Terms for LinShare along with this program. If not, see
+// <http://www.gnu.org/licenses/> for the GNU Affero General Public License version
+//  3 and <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf> for
+//  the Additional Terms applicable to LinShare software.
+//
+
+import 'dart:async';
+
+import 'package:dartz/dartz.dart';
+import 'package:dio/dio.dart';
+import 'package:domain/domain.dart';
+import 'package:domain/src/model/document/document.dart';
+import 'package:domain/src/repository/authentication/credential_repository.dart';
+import 'package:domain/src/repository/authentication/token_repository.dart';
+import 'package:domain/src/repository/document/document_repository.dart';
+import 'package:domain/src/state/failure.dart';
+import 'package:domain/src/state/success.dart';
+
+class DownloadFileIOSInteractor {
+  final DocumentRepository _documentRepository;
+  final TokenRepository _tokenRepository;
+  final CredentialRepository _credentialRepository;
+
+  DownloadFileIOSInteractor(this._documentRepository, this._tokenRepository, this._credentialRepository);
+
+  Stream<Either<Failure, Success>> execute(Document document) {
+    StreamController streamController = StreamController<Either<Failure, Success>>();
+
+    try {
+      Future.wait([_tokenRepository.getToken(), _credentialRepository.getBaseUrl()], eagerError: true)
+          .then((List responses) async =>
+          _documentRepository.downloadDocumentIOS(document, responses[0], responses[1], CancelToken())
+              .forEach((filePath) {
```

Comment:
Cool for using `Stream`. But we really have a stream of `filePaths` or only one? Please focus on the current implementation, who know about the future :)? 
> A happy man is too satisfied with the present to dwell too much on the future. - Albert Einstein.